src/uwtools/drivers/driver.py

```diff
@@ -295,8 +309,13 @@ def __init__(
             dry_run=dry_run,
             key_path=key_path,
             schema_file=schema_file,
+            coupler=coupler,
         )
         self._batch = batch
+        if coupler:
+            self._config[self._driver_name][STR.execution] = self._config_full[coupler][
+                STR.execution
+            ]
```

Comment:
If a `coupler` name was provided, set the driver config's `execution:` item from the named block. A parent class' constructor will take care of `rundir:`.